### PR TITLE
Refactor SignupReferralsGallery as SignupReferralsBlock

### PR DESCRIPTION
**Pull request by aaronschachter**
### What's this PR do?

This pull request moves the `SignupReferralsGallery` component out of the `pages/AccountPage/ReferFriends` dir and into a new `blocks/SignupReferralsBlock` tab, renaming it to `SignupReferralsBlock`. 

It also adds an `ErrorBlock` to better indicate when the `getReferFriendsLink` errors out.

### How should this be reviewed?

👀 

### Any background context you want to provide?

We'll eventually add a new `signupReferralsBlock` content type to allow placing a `SignupReferralsBlock` on a campaign action page. 

### Relevant tickets

References [Pivotal #174263179](https://www.pivotaltracker.com/story/show/174263179).

### Checklist

- [x] This PR has been added to the relevant Pivotal card.
- [ ] Documentation added for new features/changed endpoints.
- [ ] Added screenshots of front-end changes on small, medium, and large screens.
- [x] Added appropriate feature/unit tests.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -40,7 +40,6 @@ jobs:
           key: node-v2-{{ checksum "package-lock.json" }}
           paths:
             - node_modules
-            - ~/.cache/Cypress
```
Comment (aaronschachter) [inline .circleci/config.yml:43]: Testing removing this to see if this fixes the CircleCI error we started seeing today:
```
/home/circleci/.cache/Cypress/3.8.3/Cypress/Cypress: error while loading shared libraries: libXss.so.1: cannot open shared object file: No such file or directory
```
I tried running `npm install` to update the `package-lock.json`, thinking that could [clear the cache](https://github.com/DoSomething/phoenix-next/blob/9636f2781ec89cdd64e40e4bd94936e2859615b5/.circleci/config.yml#L36), but didn't have any luck in 19f5b8de3b3cdbe87960d30d76ceab0f353fe2ec.

Comment (aaronschachter) [reply .circleci/config.yml:43]: Ahh that worked! https://dosomething.slack.com/archives/CUQMTP0RM/p1598551401002200. I'm now going to try to revert this change next, as this was put here for a reason.

Comment (aaronschachter) [reply .circleci/config.yml]: Reverted this change, build succeeded. Reverting the `package-lock` updates in https://github.com/DoSomething/phoenix-next/commit/19f5b8de3b3cdbe87960d30d76ceab0f353fe2ec next.

File: resources/assets/components/pages/AccountPage/ReferFriends/ReferFriendsTab.js [hunk 2]
```diff
@@ -13,6 +14,15 @@ const DetailsParagraph = tw.p`pt-2`;
...
   const referralIncentive = featureFlag('refer_friends_incentive');
+  const referFriendsLink = getReferFriendsLink();
+
+  if (!referFriendsLink) {
+    return (
+      <div className="col-span-4 md:col-span-8 lg:col-start-2 lg:col-span-7 xxl:col-start-2 xxl:col-span-6">
+        <ErrorBlock error="Refer Friends link is undefined. Is the default_referral_campaign_id config set?" />
+      </div>
+    );
+  }
```
Comment (mendelB) [inline resources/assets/components/pages/AccountPage/ReferFriends/ReferFriendsTab.js:25]: 👏 this is a great idea!

**Pull request merged**